lib/vlc.js

```diff
@@ -1,196 +1,211 @@
-    l = recent.length;
-    // showing recent 5 songs only
-    if (l > 0) {
-      if (l >= 5) {
-        for (var a = 0; a < 4; a++) {
-          x.push(recent[a]);
+    var queuefunc = function() {
+        console.log('Queuing...');
+        playing = 0;
+        now = playing;
+        // for recent
+        x = [];
+        x.push(to_recent);
+        l = recent.length;
+        // showing recent 5 songs only
+        if (l > 0) {
+            if (l >= 5) {
+                for (var a = 0; a < 4; a++) {
+                    x.push(recent[a]);
+                }
+            } else {
+                for (var a = 0; a < l; a++) {
+                    x.push(recent[a]);
+                }
+            }
         }
-      } else {
-        for (var a = 0; a < l; a++) {
-          x.push(recent[a]);
+        recent = x;
+        //
+        if (typeof queue[0] !== 'undefined') {
+            play(queue[0][0], queue[0][1]);
+            queue.shift();
         }
-      }
     }
-    recent = x;
-    //
-    if (typeof queue[0] !== 'undefined') {
-      play(queue[0][0], queue[0][1]);
-      queue.shift();
-    }
-  }
-
-  var queuelist = function() {
-    return queue;
-  }
 
-  var recentlist = function() {
-    return recent;
-  }
+    var queuelist = function() {
+        return queue;
+    }
 
-  var _getCurrent = function() {
-    return currentTrack;
-  }
-  var rightnow = function() {
-    return now;
-  }
+    var recentlist = function() {
+        return recent;
+    }
 
-  var getYoutubeInfo = function(id) {
-    return youtubeData[id];
-  }
+    var _getCurrent = function() {
+        return currentTrack;
+    }
+    var rightnow = function() {
+        return now;
+    }
 
-  var volume = function(change) {
-    if(change == '0') {
-      var child = spawn("nircmd.exe", ["mutesysvolume", "2"]);
+    var getYoutubeInfo = function(id) {
+        return youtubeData[id];
     }
-    else {
-      var child = spawn("nircmd.exe", ["changesysvolume", change]);
+
+    var volume = function(change) {
+        if (json.env_OS == "linux") {
+            change = change + "%";
+            if (change == '0') {
+                var child = spawn("pactl", ["--", "set-sink-volume", "0", "0%"]);
+            } else {
+                var child = spawn("pactl", ["--", "set-sink-volume", "0", change]);
+            }
+        } else {
+            if (change == '0') {
+                var child = spawn("nircmd.exe", ["mutesysvolume", "2"]);
+            } else {
+                change = change * 65535 / 100;
+                var child = spawn("nircmd.exe", ["setsysvolume", change]);
+            }
+        }
     }
-  }
 
-  return {
-    play: play,
-    getCurrent: _getCurrent,
-    queuelist: queuelist,
-    kill: kill,
-    recentlist: recentlist,
-    rightnow: rightnow,
-    repeatCurrent: repeatCurrent,
-    getNext: getNext,
-    getYoutubeInfo: getYoutubeInfo,
-    volume: volume
-  }
-})();
\ No newline at end of file
+    return {
```

Comment:
Please keep the style guide as it is everywhere. Since this repository is using space instead of tabs, use them.

If possible add editorconfig to this repository. As an example use this example config: https://github.com/apsdehal/qunit-migrate/blob/master/.editorconfig